extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -160,20 +166,95 @@ export class RealTimeConfigManager {
    * @visibleForTesting
    */
   maybeExecuteRealTimeConfig(customMacros, consentState) {
-    // TODO(keithwrightbos) - allow pub to override such that some callouts are
-    // still allowed.
-    if (consentState == CONSENT_POLICY_STATE.INSUFFICIENT ||
-        consentState == CONSENT_POLICY_STATE.UNKNOWN) {
-      return;
-    }
     if (!this.validateRtcConfig_(this.a4aElement_.element)) {
       return;
     }
+    this.consentState_ = consentState;
+    this.modifyRtcConfigForConsentStateSettings();
     customMacros = this.assignMacros(customMacros);
     this.rtcStartTime_ = Date.now();
     this.handleRtcForCustomUrls(customMacros);
     this.handleRtcForVendorUrls(customMacros);
-    return Promise.all(this.promiseArray_);
+    return this.promiseArray_.length ? Promise.all(this.promiseArray_) :
+      null;
+  }
+
+  /**
+   * Returns whether a given callout object is valid to send an RTC request
+   * to, for the given consentState.
+   * @param {Object|string} callout
+   * @return {boolean}
+   */
+  isValidCalloutForConsentState(callout) {
+    if (!isObject(callout) || !callout['sendRegardlessOfConsentState']) {
+      return false;
+    }
+
+    const sendRegardlessOfConsentState =
```

Comment:
switch (typeof callout.sendRegardlessOfConsentState) {
   case 'boolean':
      return callout.sendRegardlessOfConsentState;
   case 'Array':
      return sendRegardlessOfConsentState.includes(this.consentState_);
   default:
      user().warn(TAG, 'WHAT?! ...');
      return false;
}